createdisk.sh

```diff
@@ -80,7 +80,7 @@ EOF
     ${SSH} core@${VM_IP} -- sudo systemctl disable firewalld
     # Copy the sample microshift config and update the base domain with crc base domain
     ${SSH} core@${VM_IP} -- sudo cp /etc/microshift/config.yaml.default /etc/microshift/config.yaml
-    ${SSH} core@${VM_IP} -- "sudo sed -i 's/#baseDomain: .*/baseDomain: ${SNC_PRODUCT_NAME}.${BASE_DOMAIN}/g' /etc/microshift/config.yaml"
+    ${SSH} core@${VM_IP} -- "sudo sed -i 's/baseDomain: .*/baseDomain: ${SNC_PRODUCT_NAME}.${BASE_DOMAIN}/g' /etc/microshift/config.yaml"
```

Comment:
Change this to `^baseDomain` or `^#?baseDomain` (not sure about the syntax) so that we don't regress again if this gets commented again. And I'd add a check/error if `baseDomain` was not modified as expceted.

Reply:
or just use yq/jq to edit the file and ensure we set the right value in the right place?

Reply:
> Change this to ^baseDomain or ^#?baseDomain (not sure about the syntax) so that we don't regress again if this gets commented again. 

`^baseDomain` means line start with `baseDomain` but that is not the case

> or just use yq/jq to edit the file and ensure we set the right value in the right place?

Using `yq` seems better option but VM doesn't contain yq binary it have `jq` but with `jq` we can't make modification in yaml file right?

>  And I'd add a check/error if baseDomain was not modified as expceted
yes I will add that.

Reply:
> [^baseDomain](url) means line start with baseDomain but that is not the case

https://www.gnu.org/software/sed/manual/html_node/Character-Classes-and-Bracket-Expressions.html

> Using yq seems better option but VM doesn't contain yq binary it have jq but with jq we can't make modification in yaml file right?

but we have yq on the host, could pipe guest `cat config.yaml.example` through host yq and write that to the guest

Reply:
used container image of yq so it should be good.